runtime/doc/map.txt

```diff
@@ -1431,6 +1431,223 @@ Possible values are (second column is the short name used in listing):
+		vim.o.hlsearch = false
+	    end
+
+	    -- If in command preview mode and inccommand=split,
+	    -- open the preview window
+	    -- Otherwise close preview window
+	    if cmdpreview and incsplit then
+		preview_win = vim.cmd_preview.open_win()
+		preview_buf = vim.api.nvim_win_get_buf(preview_win)
+		preview_buf_line = 0
+	    else
+		vim.cmd_preview.close_win()
+	    end
+
+	    for i, line in ipairs(lines) do
+		local startidx, endidx = string.find(line, '%s+$')
+
+		if startidx ~= nil then
+		    -- Highlight the match if in command preview mode
+		    if cmdpreview then
+			vim.api.nvim_buf_add_highlight(
+			    orig_buf,
+			    ns,
+			    'Substitute',
+			    line1 + i - 2,
+			    startidx - 1,
+			    endidx
+			)
+			-- Add lines and highlight to the preview buffer
+			-- if inccommand=split
+			if incsplit then
+			    local prefix = string.format(
+				'|%d| ',
+				line1 + i - 1
+			    )
+			    vim.api.nvim_buf_set_lines(
+				preview_buf,
+				preview_buf_line,
+				preview_buf_line,
+				0,
+				{ prefix .. line }
+			    )
+			    vim.api.nvim_buf_add_highlight(
+				preview_buf,
+				ns,
+				'Substitute',
+				preview_buf_line,
+				#prefix + startidx - 1,
+				#prefix + endidx
+			    )
+			    preview_buf_line = preview_buf_line + 1
+			end
+		    end
+		end
+
+		if not cmdpreview then
+		    new_lines[#new_lines+1] = string.gsub(
+			line,
+			'%s+$',
+			''
+		    )
+		end
+	    end
+
+	    -- Don't make any changes to the buffer if in command preview mode
+	    if not cmdpreview then
+		vim.api.nvim_buf_set_lines(
+		    orig_buf,
+		    line1 - 1,
+		    line2,
+		    0,
+		    new_lines
+		)
+	    end
+
+	    -- If preview window was opened, scroll to the top of it
+	    if cmdpreview and incsplit then
+		vim.api.nvim_win_set_cursor(preview_win, {1, 0})
+	    end
+
+	    if cmdpreview then
+		-- Redraw screen to show the highlights
+		vim.cmd('redraw')
+		-- Clear highlights
+		vim.api.nvim_buf_clear_namespace(orig_buf, ns, 0, -1)
+
+		-- If preview window was opened, switch back to the original
+		-- window to make sure that the command gets executed on that
+		-- window and not on the preview window
+		if incsplit then
+		    vim.api.nvim_set_current_win(orig_win)
+		end
+
+		-- Set previous values of 'cursorcolumn', 'cursorline'
+		-- and 'hlsearch' if they were unset
+		vim.o.cursorcolumn = save_cursorcolumn
+		vim.o.cursorline = save_cursorline
+		vim.o.hlsearch = save_hlsearch
+	    end
+	end
+
+	-- Create the user command
+	vim.api.nvim_create_user_command(
+		'TrimTrailingWhitespace',
+		TrimTrailingWhitespace,
+		{ nargs = '?', range = '%', addr = 'lines',
+		  preview = PreviewFunc })
+<
+
+Note that in this case, no actual changes to the main buffer are made for the
+preview, only the preview window and highlights are set to show the trailing
+whitespaces. It's possible to also make changes to the main buffer for the
+preview, as long as the Nvim command preview "protocol" is followed.
```

Comment:
Same as before, the word "protocol" feels out of place. Maybe something like `command preview steps` instead?